src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -886,7 +886,7 @@ private bool EmitFieldLoadReceiverAddress(BoundExpression receiver)
 
                 if (!field.IsStatic && EmitFieldLoadReceiverAddress(fieldAccess.ReceiverOpt))
                 {
-                    Debug.Assert(!field.IsVolatile, "volatile valuetype fields are unexpected");
+                    Debug.Assert(!field.IsVolatile, "volatile value type fields are unexpected");
```

Comment:
I think either way is ok. It is just an assert text and either way it is sufficiently clear what went wrong.